File: ExpenseTracker-Backend/src/main/java/com/expensetracker/ExpenseTracker/services/budget/BudgetService.java
package com.expensetracker.ExpenseTracker.services.budget;

import com.expensetracker.ExpenseTracker.dto.BudgetDTO;
import java.util.List;

public interface BudgetService {
    BudgetDTO createBudget(BudgetDTO budgetDTO);
    List<BudgetDTO> getAllBudgets();
    BudgetDTO getBudgetById(Long id);
    BudgetDTO updateBudget(Long id, BudgetDTO budgetDTO);
    void deleteBudget(Long id);
}

File: Project - Frontend/expenseTrackerWeb/src/app/services/budget/budget.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

const BASIC_URL = "http://localhost:8080/";

@Injectable({
  providedIn: 'root'
})
export class BudgetService {

  constructor(private http: HttpClient) { }

  postBudget(budgetDTO: any): Observable<any> {
      return this.http.post(BASIC_URL + "api/budgets", budgetDTO);
    }

  getAllBudgets(): Observable<any> {
      return this.http.get(BASIC_URL + "api/budgets/all");
      }

  getBudgetById(id: number): Observable<any> {
      return this.http.get(BASIC_URL + `api/budgets/${id}`);
    }

  deleteBudget(id:number): Observable<any>{
      return this.http.delete(BASIC_URL + `api/budgets/${id}`);
    }

  updateBudget(id: number, budgetDTO:any): Observable<any> {
      return this.http.put(BASIC_URL + `api/budgets/${id}`, budgetDTO);
    }
}
